Basic/DyanmicPrefabs/Assets/Scripts/04_Server Authoritative Spawn As Network-Invisible For Loading Clients/SpawnPrefabInvisibleWhileIsLoading.cs

```diff
@@ -0,0 +1,154 @@
+                {
+                    networkObjectIds.Add(obj);
+                }
+                else
+                {
+                    m_PrefabHashToNetworkObjectId.Add(assetGuid.GetHashCode(), new HashSet<NetworkObject>() {obj});
+                }
+
+                obj.CheckObjectVisibility = (clientId) => 
+                {
+                    //if the client has already loaded the prefab - we can make the object visible to them
+                    if (DynamicPrefabLoadingUtilities.HasClientLoadedPrefab(clientId, assetGuid.GetHashCode()))
+                    {
+                        return true;
+                    }
+                    //otherwise the clients need to load the prefab, and after they ack - the ShowHiddenObjectsToClient 
+                    LoadAddressableClientRpc(assetGuid, new ClientRpcParams(){Send = new ClientRpcSendParams(){TargetClientIds = new ulong[]{clientId}}});
+                    return false;
+                };
+                
+                obj.SpawnWithOwnership(m_NetworkManager.LocalClientId);
+
+                return obj;
+            }
+        }
+        
+        void ShowHiddenObjectsToClient(int prefabHash, ulong clientId)
+        {
+            if (m_PrefabHashToNetworkObjectId.TryGetValue(prefabHash, out var networkObjects))
+            {
+                foreach (var obj in networkObjects)
+                {
+                    if (!obj.IsNetworkVisibleTo(clientId))
+                    {
+                        obj.NetworkShow(clientId);
+                    }
+                }
+            }
+        }
+        
+        [ClientRpc]
+        void LoadAddressableClientRpc(AddressableGUID guid, ClientRpcParams rpcParams = default)
+        {
+            if (!IsHost)
+            {
+                Load(guid);
+            }
+
+            async void Load(AddressableGUID assetGuid)
+            {
+                Debug.Log("Loading dynamic prefab on the client...");
+                await DynamicPrefabLoadingUtilities.LoadDynamicPrefab(assetGuid, m_ArtificialDelayMilliseconds);
+                Debug.Log("Client loaded dynamic prefab");
+                AcknowledgeSuccessfulPrefabLoadServerRpc(assetGuid.GetHashCode());
+            }
+        }
+        
+        [ServerRpc(RequireOwnership = false)]
+        void AcknowledgeSuccessfulPrefabLoadServerRpc(int prefabHash, ServerRpcParams rpcParams = default)
+        {
+            Debug.Log($"Client acknowledged successful prefab load with hash: {prefabHash}");
+            DynamicPrefabLoadingUtilities.RecordThatClientHasLoadedAPrefab(prefabHash, 
+                rpcParams.Receive.SenderClientId);
+           
+            //the server has all the objects visible, no need to do anything
+            if (rpcParams.Receive.SenderClientId != m_NetworkManager.LocalClientId)
+            {
+                ShowHiddenObjectsToClient(prefabHash, rpcParams.Receive.SenderClientId);
```

Comment:
I could see a potential security issue here if users are using NetworkShow and Hide for security purposes in addition to this technique.
A malicious client could call this RPC, say "I loaded all _enemy-players-that-are-hidden_" and this method here would show them all to that client.
No need to change code, I'd just add a note?